src/shared/numa-util.c

```diff
@@ -124,6 +126,60 @@ int numa_to_cpu_set(const NUMAPolicy *policy, CPUSet *ret) {
         return 0;
 }
 
+static int numa_max_node(int *ret) {
+        _cleanup_closedir_ DIR *d = NULL;
+        struct dirent *de;
+        int r, max_node = 0;
+
+        assert(ret);
+
+        d = opendir("/sys/devices/system/node");
+        if (!d)
+                return -errno;
+
+        FOREACH_DIRENT(de, d, break) {
+                int node;
+                const char *n;
+
+                (void) dirent_ensure_type(d, de);
+
+                if (de->d_type != DT_DIR)
+                        continue;
+
+                n = startswith(de->d_name, "node");
+                if (!n)
+                        continue;
+
+                r = safe_atoi(n, &node);
+                if (r < 0)
+                        return r;
```

Comment:
mabye "continue" would be better than "return r" here, too, since kernel peeps might come up with new subdirs here, maybe "node_table" or "nodes" or "node_current" or so, and they all would start with "node" but might be something differnt.